tests/sles4sap/robot_fw.pm

```diff
@@ -32,6 +33,7 @@ sub run {
     # Install the robot framework
     assert_script_run "unzip /robot/bin/robotframework-$robot_fw_version.zip";
     assert_script_run "cd robotframework-$robot_fw_version";
+    zypper_call "in $python_bin-setuptools" if (script_run "rpm -qa | grep $python_bin-setuptools");
```

Comment:
Is it really necessary to test if the package is not already installed? AFAIK `zypper_call` will not fail if the package already exists. And with `script_run` it's also better to test if the result is defined IMO.

Reply:
I did that before, the purpose is to avoid a zypper command with refresh etc... when it is not needed.

Reply:
And `rpm -q $python_bin-setuptools` is also better, avoid unneeded `grep` ;-)

Reply:
I also think `rpm -q $python_bin-setuptools` would be better than the current command with grep.

WRT the output of `script_run`, it is going to be either 0 (package is already installed), 1 (package is not installed) or undef (timeout or other error while interacting with SUT). So IMO current code with `if (script_run "rpm -q $python_bin-setuptools");` should be good enough as it would expect for an explicit true value (non-zero / non-undef) to install the package. Only problem would be if it returns undef, in which case it will not install the package and continue ... and then fail in L37.

Reply:
Thanks for your opinions, finally I'm using `is_package_installed` (Thanks Loïc, I didn't think about this)